java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -123,25 +127,69 @@ private static Field getUnionField(Schema schema, String name) {
     int size = schema.getTypes().size();
     long nullCount = schema.getTypes().stream().filter(s -> s.getType() == Type.NULL).count();
 
-    // avro schema not allow repeated type, so size == nullCount + 1 indicates nullable type.
-    if (size == nullCount + 1) {
+    // union only has one type, convert to primitive type
+    if (size == 1) {
+      Schema subSchema = schema.getTypes().get(0);
+      return getArrowField(subSchema, name,true);
```

Comment:
```suggestion
      return getArrowField(subSchema, name, /*nullable=*/false);
```
Shouldn't a single value be not nullable (unless the type in the union is null)?  

I'm also not sure if we should collapse here always, it seems like a reasonable option for now, but it would prevent round-tripping (not that that is a use-case at the momement).

Reply:
Thanks, fixed. Not sure if we should implement a NullVector to hold null values(ZeroVector length always 0).

>I'm also not sure if we should collapse here always, it seems like a reasonable option for now, but it would prevent round-tripping (not that that is a use-case at the momement).

Not quite understand here :), but seems not a blocker now.

Reply:
> Not sure if we should implement a NullVector
We should do this and add it to the integration tests, the other missing type which might have made this easier is a DenseUnion representation.